nixos/modules/system/etc/setup-etc.pl

```diff
@@ -1,82 +1,111 @@
-use strict;
-use File::Find;
-use File::Copy;
-use File::Path;
-use File::Basename;
-use File::Slurp;
-
-my $etc = $ARGV[0] or die;
+#! @perl@/bin/perl
+
+use v5.34;
+use warnings;
+use experimental qw(try signatures);
+
+use File::Basename qw(dirname);
+use File::Copy qw(copy);
+use File::Find qw(find);
+use File::Path qw(make_path rmtree);
+use File::Slurp qw(read_file write_file read_dir);
+
+## no critic(RegularExpressions::RequireLineBoundaryMatching)
+## no critic(RegularExpressions::RequireExtendedFormatting)
+## no critic(RegularExpressions::RequireDotMatchAnything)
+
+my $etc = $ARGV[0];
+if ($etc eq "") {
+    die("This script must be called with an argument which expresses the new /etc directory in the nix store");
+}
 my $static = "/etc/static";
```

Comment:
I think this might be better off as `use constant STATIC => "/etc/static";` to set it apart as a constant both here at the top of the file and where it is used

Reply:
Not sure if this really makes it better perl code as i am not an expert, i added it anyways, thanks for the review:)

Reply:
I'm hardly a perl expert either, but I always appreciate being able to see at a glance which name refers to a constant and which to a variable. Thanks for adopting the change :)